streams/src/main/java/org/apache/kafka/streams/processor/StreamTaskMetadata.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.processor;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+public class StreamTaskMetadata extends TaskId {
```

Comment:
I agree with @vvcephei comment that there is no reason that `StreamTaskMetadata` should inherit from `TaskId`.

Not even sure, if TaskId should be a field within `StreamTaskMetadata` -- I guess it can be a field, but I don't see a strong reason why it must be. Thus, IMHO we can do it either way; whatever gives cleaner code.

Also, rename from `StreamTaskMetadata` to `TaskMetadata`?

Reply:
Oh, there is already a class named ```TaskMetadata```. I had to create a new class to avoid ambiguity regarding names.

Reply:
I see. I just double checked and realized that the existing `TaskMetadata` is part of public API. The class you add, should not be part of public API and thus, it should not be added into a public package. It should rather be added to `.../processor/internals/` -- this way, we can rename to `TaskMetadata` :)